src/simulations/set_generator.py

```diff
@@ -626,3 +626,96 @@ def __iter__(self):
       self.random_state.shuffle(multiset_ids)
       yield multiset_ids
     return self
+
+class HeterogeneousMultiSetGenerator(SetGeneratorBase):
+  """Heterogeneous multiset generator.
+
+  This generator returns the multisets as described in section
+  Frequency scenario 1: Heterogeneous user frequency of this doc:
+
+  https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/doc/cardinality_and_frequency_estimation_evaluation_framework.md#frequency-scenario-2-heterogeneous-user-frequency
+  Heterogeneous means that the reached ID's have different frequency
+  distribution.
+
+  As a reached ID may appear one or more times in the returned set,
+  to differentiate with a normal set whose elements can only appear once, we
+  use the term multiset.
+
+  The frequency distribution is defined as a shifted-Poisson distribution.
+  I.e., freq ~ Poission(freq_rate) + 1.
+  """
+
+  @classmethod
+  def get_generator_factory_with_num_and_size(cls, universe_size, num_sets,
+                                              set_size, gamma_params_list,
+                                              freq_cap):
+
```

Comment:
Can you add a check for  whether the expected set_size_list == gamma_params_list here? Otherwise this error check and subsequent failure are significantly delayed. 

The stack trace you get from the error in __init__ will not easily lead you to you problem. The stack trace from here will.

Reply:
Done.